web/components/scroll-to-top-button.tsx

```diff
@@ -0,0 +1,42 @@
+import { ArrowUpIcon } from '@heroicons/react/solid'
+import clsx from 'clsx'
+import { useEffect, useState } from 'react'
+import { Row } from './layout/row'
+
+export function ScrollToTopButton(props: { className?: string }) {
+  const { className } = props
+  const [visible, setVisible] = useState(false)
+
+  useEffect(() => {
+    window.addEventListener('scroll', () => {
+      if (window.scrollY > 500) {
+        setVisible(true)
+      } else {
+        setVisible(false)
+      }
+    })
```

Comment:
This is a decent solution, but scroll event listeners are kind of slow.

(Also, you should remove the event listener when the component is destroyed, which you would do by returning a function in the useEffect: https://bobbyhadz.com/blog/react-remove-event-listener)

Another solution is to add our component `<VisibilityObserver />` at the top of the page, and show this button whenever it is not visible.